src/app/courses/add-courses/courses-add.component.ts

```diff
@@ -231,7 +231,7 @@ export class CoursesAddComponent implements OnInit, OnDestroy {
   }
 
   navigateBack() {
-    this.router.navigate([ '/courses' ]);
+    this.router.navigate([ `../../${this.courseId}` ], { relativeTo: this.route });
```

Comment:
This should not rely on the `courseId` because the routing should go back to the courses list from the `update/:id` route.